### PR TITLE
feat: remove TBD from filter syntax, some docs improvements, changing timestamp to use protobuf timestamp

### PR DESCRIPTION
**Pull request by himank**

### CODE REVIEW

File: server/v1/api.proto [hunk 1]
```diff
@@ -151,8 +151,13 @@ message DeleteRequest {
   // Collection name where to insert documents
   string collection = 2;
   // Delete documents which matching specified filter.
-  // The filter should proper JSON object.
-  // Filter syntax described here: {TBD}
+  // The filter should proper JSON object. A filter can simply be key, value where key
```
Comment (ovaistariq) [inline server/v1/api.proto:154]: nit: Remove "The filter should proper JSON object". You are already explaining well in next sentences.

File: server/v1/api.proto [hunk 2]
```diff
@@ -199,8 +209,13 @@ message ReadRequest {
   // Collection name to read documents from
   string collection = 2;
   // Returns documents matching this filter.
-  // The filter should proper JSON object
-  // Filter syntax described here: {TBD}
+  // The filter should proper JSON object. A filter can simply be key, value where key
```
Comment (ovaistariq) [inline server/v1/api.proto:212]: Same as above - remove "The filter should proper JSON object."

File: server/v1/api_openapi.yaml [hunk 3]
```diff
@@ -490,7 +490,7 @@ components:
             properties:
                 filter:
                     type: object
-                    description: 'Delete documents which matching specified filter. The filter should proper JSON object. Filter syntax described here: {TBD}'
+                    description: 'Delete documents which matching specified filter. The filter should proper JSON object. A filter can simply be key, value where key is the field name and value would be the value for this field. Or a filter can be logical where two or more fields can be logically joined using $or and $and. A few examples of filter,   1. To delete a user document where the id has a value 1.      {"id": 1 }   2. To delete all the user documents where the key "id" has a value 1 or 2 or 3.      {"$or": [{"id": 1}, {"id": 2}, {"id": 3}]}'
```
Comment (ovaistariq) [inline server/v1/api_openapi.yaml:493]: Remove "The filter should proper JSON object."

File: server/v1/api_openapi.yaml [hunk 4]
```diff
@@ -588,7 +588,7 @@ components:
             properties:
                 filter:
                     type: object
-                    description: 'Returns documents matching this filter. The filter should proper JSON object Filter syntax described here: {TBD}'
+                    description: 'Returns documents matching this filter. The filter should proper JSON object. A filter can simply be key, value where key is the field name and value would be the value for this field. Or a filter can be logical where two or more fields can be logically joined using $or and $and. A few examples of filter,   1. To read a user document where the id has a value 1.      {"id": 1 }   2. To read all the user documents where the key "id" has a value 1 or 2 or 3.      {"$or": [{"id": 1}, {"id": 2}, {"id": 3}]}'
```
Comment (ovaistariq) [inline server/v1/api_openapi.yaml:591]: Remove "The filter should proper JSON object."

File: server/v1/api_openapi.yaml [hunk 5]
```diff
@@ -676,7 +676,7 @@ components:
                     description: Fields contains set of fields with the values which need to be updated. Should be proper JSON object.
                 filter:
                     type: object
-                    description: 'Update documents which matching specified filter. The filter should proper JSON object Filter syntax described here: {TBD}'
+                    description: 'Update documents which matching specified filter. The filter should proper JSON object. A filter can simply be key, value where key is the field name and value would be the value for this field. Or a filter can be logical where two or more fields can be logically joined using $or and $and. A few examples of filter,   1. To update a user document where the id has a value 1.      {"id": 1 }   2. To update all the user documents where the key "id" has a value 1 or 2 or 3.      {"$or": [{"id": 1}, {"id": 2}, {"id": 3}]}'
```
Comment (ovaistariq) [inline server/v1/api_openapi.yaml:679]: Remove "The filter should proper JSON object."

**Review by ovaistariq — APPROVED** [review]

File: server/v1/api.proto [hunk 6]
```diff
@@ -27,98 +28,104 @@ message ErrorDetails {
...
- */
+// Contains ID which uniquely identifies transaction
+// This context is returned by BeginTransaction request and
+// should be passed in the subsequent requests in order to run
+// them in the context of the same transaction.
 message TransactionCtx {
+  // Unique for a single transactional request.
   string id = 1;
+  // Serves as an internal identifier.
   string origin = 2;
```
Comment (JigarJoshi) [inline server/v1/api.proto:70]: should we encode this into one so we hide internal things?

File: server/v1/api.proto [hunk 7]
```diff
@@ -271,47 +287,47 @@ message CreateOrUpdateCollectionRequest {
...
   string status = 2;
 }
 
 message DatabaseInfo {
-  // Database name
+  // Database name.
   string db = 1;
 
-  // Database metadata
+  // Metadata about the database. (ex. ...)
```
Comment (JigarJoshi) [inline server/v1/api.proto:322]: remove (ex. ....)

File: server/v1/api.proto [hunk 8]
```diff
@@ -271,47 +287,47 @@ message CreateOrUpdateCollectionRequest {
...
   string collection = 2;
   CollectionOptions options = 3;
 }
 
 message DropCollectionResponse {
-  // a detailed response message.
+  // A detailed response message.
   string message = 1;
-  // an enum with value set as "dropped"
+  // An enum with value set as "dropped".
```
Comment (JigarJoshi) [inline server/v1/api.proto:314]: should we switch to enum type here if there are limited set of statuses and in general.

**Review by JigarJoshi — APPROVED** [review]

File: server/v1/api.proto (hunk 8 above)
Comment (himank) [reply server/v1/api.proto:314]: that's what I was doing first but the open-api generator failed so switched back to the string.

File: server/v1/api.proto (hunk 6 above)
Comment (himank) [reply server/v1/api.proto:70]: yeah, we can do that or just encode as store encoded value here.

File: server/v1/api.proto (hunk 8 above)
Comment (JigarJoshi) [reply server/v1/api.proto:314]: ok

**Pull request merged**